shell/renderer/api/atom_api_context_bridge.cc

```diff
@@ -0,0 +1,426 @@
+  DCHECK(store->HasReverseBindingStore());
+
+  return store->GetReverseBindingsStore(dict.isolate()).Has(key);
+}
+
+void ExposeAPIInMainWorld(const std::string& key,
+                          mate::Dictionary api,
+                          base::Optional<mate::Dictionary> options,
+                          mate::Arguments* args) {
+  auto* render_frame = GetRenderFrame(api.GetHandle());
+  RenderFramePersistenceStore* store = GetOrCreateStore(render_frame);
+  DCHECK(store->HasReverseBindingStore());
+  auto* frame = render_frame->GetWebFrame();
+  DCHECK(frame);
+  v8::Local<v8::Context> main_context = frame->MainWorldScriptContext();
+  mate::Dictionary global(main_context->GetIsolate(), main_context->Global());
+
+  if (global.Has(key)) {
+    args->ThrowError(
+        "Cannot bind an API on top of an existing property on the window "
+        "object");
+    return;
+  }
+
+  v8::Local<v8::Context> isolated_context =
+      frame->WorldScriptContext(api.isolate(), World::ISOLATED_WORLD);
+
+  bool allow_reverse_binding = true;
+  if (options.has_value())
+    options->Get("allowReverseBinding", &allow_reverse_binding);
+
+  if (allow_reverse_binding && api.Has("reverseBinding")) {
+    args->ThrowError(
+        "Cannot define property 'reverseBinding' when 'allowReverseBinding' is "
+        "enabled");
+    return;
+  }
+
+  {
+    v8::Context::Scope main_context_scope(main_context);
+    mate::Dictionary proxy =
+        CreateProxyForAPI(api, isolated_context, main_context, store);
+    if (allow_reverse_binding) {
+      mate::Dictionary reverse_bindings =
+          mate::Dictionary::CreateEmpty(api.isolate());
+
+      reverse_bindings.Set(
+          "bind",
+          mate::ConvertToV8(
+              api.isolate(),
+              base::BindOnce(&ExposeReverseBindingInIsolatedWorld, key)));
+      reverse_bindings.SetMethod(
+          "isBound", base::BindRepeating(&IsReverseBindingBound, key));
+      proxy.Set("reverseBinding", reverse_bindings);
+    }
+    DeepFreeze(proxy.GetHandle(), main_context);
+    global.SetReadOnlyNonConfigurable(key, proxy);
+  }
+}
+
+void SetReverseBindingStore(v8::Isolate* isolate,
+                            v8::Local<v8::Value> reverse_bindings_store) {
+  DCHECK(reverse_bindings_store->IsObject() &&
+         !reverse_bindings_store->IsNullOrUndefined());
+  RenderFramePersistenceStore* store =
+      GetOrCreateStore(GetRenderFrame(reverse_bindings_store));
+  store->SetReverseBindingStore(
+      isolate, v8::Local<v8::Object>::Cast(reverse_bindings_store));
+}
+
+}  // namespace api
+
+}  // namespace electron
+
+namespace {
+
+void Initialize(v8::Local<v8::Object> exports,
+                v8::Local<v8::Value> unused,
+                v8::Local<v8::Context> context,
+                void* priv) {
+  using namespace electron::api;  // NOLINT(build/namespaces)
+
+  v8::Isolate* isolate = context->GetIsolate();
+  mate::Dictionary dict(isolate, exports);
```

Comment:
Can you use `gin` APIs in this file?

`mate::Dictionary` is replaced by `gin_helper::Dictionary` in `shell/common/gin_helper/dictionary.h`. Other converters for gin are defined in `shell/common/gin_converters`.

Reply:
@zcbenz My intention is to request a `6.1` and backport this PR.  To avoid churn I chose to use `mate` in this file to ease the backport.  I'll refactor to `gin` once it's landed 👍

Reply:
This is reasonable, I'm good with `mate` in this file then.